data/RecordAction.js

```diff
@@ -48,6 +52,15 @@ export class RecordAction {
      *          Can specify 0 to only enable action if no records are active.
      *      true: specifies that number of records > 0. Allows for arbitrary number of records.
      *      false: specifies any number of records (0 - infinity, inclusive). Always active.
+     * @param {(string|boolean)} [c.confirm] - whether the action needs to be confirmed before being
+     *      executed. Truthy values will determine the message displayed to the user. If provided the
+     *      action text will be used as the confirm button text and dialog title, otherwise "Confirm"
+     *      will be displayed.
+     *
+     *      string: specifies the exact message to display to the user
```

Comment:
My first thought is that adding the confirm here --- and trying to standardize the language it contains, is going one step too far.  What if a user needs multiple confirms?  data driven confirms? etc.